sys/include/net/ng_ipv6/nc.h

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @defgroup    net_ng_ipv6_nc  IPv6 neighbor cache
+ * @ingroup     net_ng_ipv6
+ * @brief       Translates IPv6 addresses to link layer addresses.
+ * @{
+ *
+ * @file
+ * @brief       Neighbor cache definitions.
+ *
+ * @author      Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+
+#ifndef NG_IPV6_NC_H_
+#define NG_IPV6_NC_H_
+
+#include <stdint.h>
+
+#include "kernel_types.h"
+#include "net/ng_ipv6/addr.h"
+#include "net/ng_netif.h"
+#include "net/ng_pktqueue.h"
+#include "timex.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   The size of the neighbor cache
+ */
+#define NG_IPV6_NC_SIZE             (NG_NETIF_NUMOF * 8)
+
+/**
+ * @brief   The maximum size of a link layer address
+ */
+#define NG_IPV6_L2_ADDR_MAX         (8)
```

Comment:
Again, it might make sense to give other modules the option to overrule this. Think of some weird radio device with huge L2 address that might want to overwrite this.